### PR TITLE
NXP-29988 add ftests cold storage [2021]

### PR DESCRIPTION
**Pull request by PGoncalvesNX**

### CODE REVIEW

**Comment by RSalem07** [comment]
@PGoncalvesNX thanks for the work 👍 

I think you can squash the two commits:
- NXP-29988: remove common-lib.groovy file
- NXP-29988: run ftests in ci

As both of them are the CI part. 

I'll continue the review asap

File: nuxeo-coldstorage-web/ftest/features/coldstorage.feature [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    And I send the file to cold storage
+    Then I can see the file is stored in cold storage
+
+  Scenario: User with ReadWrite permission can't send file to cold storage
+    Given I login as "Jack"
+    And I have a File document
+    And This document has file "sample.png" for content
+    And I have permission ReadWrite for this document
+    When I browse to the document
+    Then I cant send the file to cold storage
```
Comment (RSalem07) [inline nuxeo-coldstorage-web/ftest/features/coldstorage.feature:29]: I think the scenario is to check the not availability to send content into Cold Storage for a not granted user, I mean a user without a `WriteColdStorage` a given document. But here `Jack` already has the permission that come from the second scenario `User with WriteColdStorage permission can send file to cold storage`

File: nuxeo-coldstorage-web/ftest/features/coldstorage.feature [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    And I select the "sample1" document
+    And I select the "sample2" document
+    And I select the "sample3" document
+    And I move the files to cold storage
+    When I browse to the document with path "/default-domain/ws/sample1"
+    Then I can see the file is stored in cold storage
+    When I browse to the document with path "/default-domain/ws/sample2"
+    Then I can see the file is stored in cold storage
+    When I browse to the document with path "/default-domain/ws/sample3"
+    Then I can see the file is stored in cold storage
```
Comment (RSalem07) [inline nuxeo-coldstorage-web/ftest/features/coldstorage.feature:54]: I am wondering if we have to check we have the popup before to validate the move to cold storage.

File: nuxeo-coldstorage-web/ftest/features/coldstorage.feature (hunk 1 above)
Comment (PGoncalvesNX) [reply nuxeo-coldstorage-web/ftest/features/coldstorage.feature:29]: The scenarios are independent so Jack doesn't have the WriteColdStorage permission set in 2nd scenario.

File: nuxeo-coldstorage-web/ftest/features/coldstorage.feature (hunk 2 above)
Comment (PGoncalvesNX) [reply nuxeo-coldstorage-web/ftest/features/coldstorage.feature:54]: What pop-up? The confirmation dialog?

File: ci/jenkinsfiles/release.groovy [hunk 3]
```diff
@@ -264,6 +264,60 @@ pipeline {
         }
       }
     }
+    stage('Deploy ColdStorage Preview') {
```
Comment (RSalem07) [inline ci/jenkinsfiles/release.groovy:267]: a question more for @doulba :) 

Do we have to copy past the same code. I mean one for the common build and a second for the release?

thx

File: ci/jenkinsfiles/common-lib.groovy [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+ * - VERSION
+ */
+void skaffoldBuild(String skaffoldFile) {
+  sh """
+    envsubst < ${skaffoldFile} > ${skaffoldFile}~gen
+    skaffold build -f ${skaffoldFile}~gen
+  """
+}
+
+return this;
```
Comment (RSalem07) [inline ci/jenkinsfiles/common-lib.groovy]: sorry @PGoncalvesNX perhaps I am missing somethings, why we need this file. I mean we have already the unit test on front end part on the LTS-2021

**Review by RSalem07 — COMMENTED** [review]
Thanks for the work 👍 Pedro :). I left some comments/interrogations

File: ci/jenkinsfiles/common-lib.groovy (hunk 4 above)
Comment (PGoncalvesNX) [reply ci/jenkinsfiles/common-lib.groovy]: It was already removed before your comment 😅

File: nuxeo-coldstorage-web/ftest/features/coldstorage.feature (hunk 2 above)
Comment (RSalem07) [reply nuxeo-coldstorage-web/ftest/features/coldstorage.feature:54]: yes,

File: nuxeo-coldstorage-web/ftest/features/coldstorage.feature (hunk 1 above)
Comment (RSalem07) [reply nuxeo-coldstorage-web/ftest/features/coldstorage.feature:29]: ah ok, but why Jack  is still able to move into cold storage `Then I cant send the file to cold storage`

File: nuxeo-coldstorage-web/ftest/features/coldstorage.feature (hunk 2 above)
Comment (PGoncalvesNX) [reply nuxeo-coldstorage-web/ftest/features/coldstorage.feature:54]: That is included inside the steps `I move the files to cold storage` and `I send the file to cold storage`.
```
  const dialog = this.ui.browser.el.element('nuxeo-dialog#contentsToMoveDialog');
  dialog.waitForVisible();
  dialog.click('paper-button[name="confirm"]');
```

File: nuxeo-coldstorage-web/ftest/features/coldstorage.feature (hunk 1 above)
Comment (PGoncalvesNX) [reply nuxeo-coldstorage-web/ftest/features/coldstorage.feature:29]: The step says "I cant send". It's to validate that he can't see the action button.

File: ci/jenkinsfiles/release.groovy (hunk 3 above)
Comment (doulba) [reply ci/jenkinsfiles/release.groovy:267]: Hello @RSalem07 
Unfortunately having shared libs don't avoid to duplicate stage name

Comment (doulba) [reply ci/jenkinsfiles/release.groovy:267]: commons libs file will be deleted by Pedro

File: nuxeo-coldstorage-web/ftest/features/coldstorage.feature (hunk 1 above)
Comment (RSalem07) [reply nuxeo-coldstorage-web/ftest/features/coldstorage.feature:29]: sorry, but it's a little confusing for me. Is it possible to create a step saying `I cannot see the send to cold storage button`. My idea is to be more consistent with the scenario name.

Comment (PGoncalvesNX) [reply nuxeo-coldstorage-web/ftest/features/coldstorage.feature:29]: Sure

File: nuxeo-coldstorage-web/ftest/features/coldstorage.feature (hunk 2 above)
Comment (RSalem07) [reply nuxeo-coldstorage-web/ftest/features/coldstorage.feature:54]: yes, I know I saw it. But is it possible to add a specific scenario to check its presence. Thanks

**Review by RSalem07 — COMMENTED** [review]
Thanks Pedro for the work. I left two small :) suggestions

File: nuxeo-coldstorage-web/ftest/features/coldstorage.feature (hunk 2 above)
Comment (PGoncalvesNX) [reply nuxeo-coldstorage-web/ftest/features/coldstorage.feature:54]: There is no scenario in which the confirmation dialog appears without clicking on the action first, and there is no scenario in which I click on the action and I don't expect the dialog to appear. I don't understand why we need a specific step for the confirmation.

Comment (RSalem07) [reply nuxeo-coldstorage-web/ftest/features/coldstorage.feature:54]: What I had in mind is the check the popup dialog behaviour.
- Visible -> for example the scenrio will be: 

```
Scenario: User with WriteColdStorage permission must see popup dialogue confirmation before sending file to cold storage
    Given I login as "Jack"
    And I have a File document
    And This document has file "sample.png" for content
    And I have permission WriteColdStorage for this document
    When I browse to the document
    And I click the action button 
    Then I should see a pop dialogue for confirmation
```
thing like that, it's like what you did for Aspera. 



- Accept to send the content (exactly what you did)
- Refuse, then the no move to cold storage is done. 


I am not talking on the action itself, I am talking about the dialogue once you click the button / action.

Comment (PGoncalvesNX) [reply nuxeo-coldstorage-web/ftest/features/coldstorage.feature:54]: Ah, ok. So you want me to add a scenario where I cancel the move to coldstorage in the confirmation dialog. That wasn't what I understood at first.

Comment (RSalem07) [reply nuxeo-coldstorage-web/ftest/features/coldstorage.feature:54]: > Ah, ok. So you want me to add a scenario where I cancel the move to coldstorage in the confirmation dialog.
yep, thanks :) 

> That wasn't what I understood at first.
sorry about that. 



Thanks again for your efforts

File: nuxeo-coldstorage-web/ftest/features/coldstorage.feature (hunk 1 above)
Comment (RSalem07) [reply nuxeo-coldstorage-web/ftest/features/coldstorage.feature]: Thanks 👍

File: nuxeo-coldstorage-web/ftest/features/coldstorage.feature [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+  Scenario: User without WriteColdStorage permission to a document cannot see the Send the selected files to cold storage action when said document is selected in Workspace view
```
Comment (RSalem07) [inline nuxeo-coldstorage-web/ftest/features/coldstorage.feature:69]: hey Perdo, thx for taking into account the code review.
 nice to have -> can we rename a little bit the scenario name for example I was thinking to something like:

`User without WriteColdStorage permission on all selected documents cannot send them to cold storage`

Comment (PGoncalvesNX) [reply nuxeo-coldstorage-web/ftest/features/coldstorage.feature:69]: Thanks for the suggestion! I struggled to get a good name for the scenario and also think it ended up being too long 😄

File: ci/jenkinsfiles/build.groovy [hunk 6]
```diff
@@ -260,6 +247,52 @@ pipeline {
...
+                )
+              }
+            } catch(err) {
+              throw err
+            } finally {
+              //retrieve preview logs
+              nxKube.helmGetPreviewLogs("${PREVIEW_NAMESPACE}")
+              cucumber (
+                fileIncludePattern: '**/*.json',
+                jsonReportDirectory: "${FRONTEND_FOLDER}/target/cucumber-reports/",
```
Comment (doulba) [inline ci/jenkinsfiles/build.groovy:268]: Hello @PGoncalvesNX 
the report wasn't generated
Screenshot 2021-04-09 at 13 50 50

Comment (PGoncalvesNX) [reply ci/jenkinsfiles/build.groovy:268]: What do I need to change in the code?

Comment (doulba) [reply ci/jenkinsfiles/build.groovy:268]: > What do I need to change in the code?

perhaps , we should update the content of the package.json file
but I'm not sure 🤔 ( do you have the report when running the test locally ?)

Comment (PGoncalvesNX) [reply ci/jenkinsfiles/build.groovy:268]: Yes, I do 😬

**Review by lisamcintyre — APPROVED** [review]
LGTM.  I think we've captured the needs around sending and can use this as a basis for our retrieval tests, too.

**Review by doulba — APPROVED** [review]

**Review by RSalem07 — APPROVED** [review]

**Pull request merged**